test/common/router/config_impl_test.cc

```diff
@@ -1651,52 +1686,116 @@ TEST_F(RouterMatcherHashPolicyTest, HashIpv6DifferentAddresses) {
   }
 }
 
+TEST_F(RouterMatcherHashPolicyTest, QueryParameterHash) {
+  firstRouteHashPolicy()->mutable_query_parameter()->set_key("hash");
+  uint64_t hash_1;
+  uint64_t hash_2;
+  {
+    Http::TestHeaderMapImpl headers = genHeaders("www.lyft.com", "/foo?hash=bluh", "GET");
+    const auto hash_policy = config().route(headers, 0)->routeEntry()->hashPolicy();
+    hash_1 = hash_policy->generateHash(nullptr, headers, add_cookie_nop_).value();
+  }
+  {
+    Http::TestHeaderMapImpl headers = genHeaders("www.lyft.com", "/foo?hash=meh", "GET");
+    const auto hash_policy = config().route(headers, 0)->routeEntry()->hashPolicy();
+    hash_2 = hash_policy->generateHash(nullptr, headers, add_cookie_nop_).value();
+  }
+  EXPECT_NE(hash_1, hash_2);
+}
+
+TEST_F(RouterMatcherHashPolicyTest, QueryParameterHashNoSuchKey) {
```

Comment:
For completeness, can you add a version of this test without any query parameters specified in the URL?

Reply:
ah that's missing, will add it if we add back QueryParam hash method.